celery/worker/consumer/consumer.py

```diff
@@ -305,7 +306,7 @@ def _limit_task(self, request, bucket, tokens):
 
     def start(self):
         blueprint = self.blueprint
-        while blueprint.state != CLOSE:
+        while blueprint.state not in {CLOSE, TERMINATE}:
```

Comment:
If you allow me a suggestion here, I would propose to organize the shutdown states in a single point:

```python
shutdown_states = {CLOSE, TERMINATE}
while blueprint.state not in shutdown_states:
    ...
    if blueprint.state not in shutdown_states:
```

I believe that way it will be easier to avoid regressions in the future by modifying the states set in one of the state checks. What do you think?